console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,50 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsolePlugin is an extension for customizing OpenShift web console by
+// dynamically loading code from another service running on the cluster.
+type ConsolePlugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec ConsolePluginSpec `json:"spec"`
+}
+
+// ConsolePluginSpec is the desired plugin configuration.
+type ConsolePluginSpec struct {
+	// displayName is the display name of the plugin.
+	DisplayName string `json:"displayName"`
+	// service is a Kubernetes Service that exposes the plugin using a
+	// deployment with an HTTP server. The Service must use HTTPS and
+	// service serving certificate. The console backend will proxy the
+	// plugins assets from the Service using the service CA bundle.
+	Service ConsolePluginService `json:"service"`
+}
+
+// ConsolePluginService holds informations on service that is serving
+// dynamic plugin for the console.
+type ConsolePluginService struct {
+	// name of Service that is serving the plugin.
+	Name string `json:"name"`
+	// namespace of Service that is serving the plugin.
+	Namespace string `json:"namespace"`
+	// port on which the Service that is serving the plugin is listening to.
+	Port int32 `json:"port"`
+	// manifest is the path to the plugin manifest. The manifest is a JSON
+	// document that contains metadata about the plugin and the extensions.
+	Manifest string `json:"manifest"`
```

Comment:
The name of Console dynamic plugin manifest file (`plugin-manifest.json`) is fixed.

In the above data structure, `name` + `namespace` + `port` are used to resolve the plugin service URL address. Console web application refers to this address as "plugin base URL".

Given the above, I think that the `manifest` field is redundant.

Reply:
Is there a reason to force a specific path for the manifest file, particularly when it's so easy to make it a property? You might want to serve your plugin from a service that does other things as well and move the plugin assets under a specific directory as an example.

Reply:
I talked to @vojtechszocs. I'm good changing this to something like `basePath` and having the manifest always be at `<basePath>/plugin-manifest.json`. But either way we need some property here to let plugins have a context root for the plugin and not assume everything can be served at the base path.

Reply:
```suggestion
	BasePath string `json:"manifest"`
```

Reply:
> But either way we need some property here to let plugins have a context root for the plugin and not assume everything can be served at the base path.

Not sure why that can't be handled by the `BasePath` ?
Wont the plugin server should be handling everything on the `<endpoint>:<port>/<basepath>/*` ?